dlpy/model_conversion/write_onnx_model.py

```diff
@@ -0,0 +1,699 @@
+#!/usr/bin/env python
+# encoding: utf-8
+#
+# Copyright SAS Institute
+#
+#  Licensed under the Apache License, Version 2.0 (the License);
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+''' Write ONNX model '''
+
+from onnx import helper, numpy_helper
+from onnx import TensorProto
+import numpy as np
+
+
+class OnnxWriteError(ValueError):
+    '''
+    Used to indicate an error in parsing ONNX model definition
+
+    '''
+
+
+def sas_to_onnx(layers, model_table, model_weights):
+    '''
+    Convert DLPy model to ONNX
+
+    Parameters
+    ----------
+    layers : iter-of-Layers
+        Specifies the layers defining the model.
+    model_table : :class:`CASTable`
+        Specifies the CASTable of the model.
+    model_weights : :class:`pandas.DataFrame` or :class:`CASTable`
+        DataFrame or CASTable containing the model weights.
+        If this is a CASTable, the weights will be fetched from
+        the CAS server.  This may take a long time if 
+        the model has many weights.
+
+    Returns
+    -------
+    Loaded in-memory ModelProto
+
+    '''
+    nodes = []
+    inputs = []
+    outputs = []
+    initializer = []
+    
+    import pandas as pd
+    if isinstance(model_weights, pd.DataFrame):
+        fetch = False
+    else:
+        fetch = True
+
+    model_name = model_table.query('_DLKey1_ = "modeltype"') \
+                            .fetch()['Fetch']['_DLKey0_'][0]
+
+    for layer in layers:
+        if layer.__class__.__name__.lower() == 'inputlayer':
```

Comment:
there should be something like this layer.type() == 'inputlayer', can you check it out?

Reply:
changed to use layer.type